docs/src/testing/using-cheatcodes.md

```diff
@@ -15,45 +16,68 @@ using [cheatcodes](../appendix/cheatcodes.md).
 
 ## The Test Contract
 
-In this tutorial will be using this Starknet contract:
+In this tutorial, we will be using the following Starknet contract:
 
 ```rust
 #[starknet::interface]
 trait IHelloStarknet<TContractState> {
     fn increase_balance(ref self: TContractState, amount: felt252);
     fn get_balance(self: @TContractState) -> felt252;
+    fn get_block_number(self: @TContractState) -> u64;
+    fn get_block_timestamp(self: @TContractState) -> u64;
 }
 
 #[starknet::contract]
 mod HelloStarknet {
+
+    use box::BoxTrait;
+    use starknet::{Into, get_caller_address};
+
     #[storage]
     struct Storage {
         balance: felt252,
+        blk_nb: u64,
+        blk_timestamp: u64,
+    }
+
+    #[constructor]
+    fn constructor(ref self: ContractState) {
+        // store the current block number
+        self.blk_nb.write(starknet::get_block_info().unbox().block_number);
+        // store the current block timestamp
+        self.blk_timestamp.write(starknet::get_block_info().unbox().block_timestamp);
     }
 
     #[external(v0)]
-    impl HelloStarknetImpl of super::IHelloStarknet<ContractState> {
+    impl IHelloStarknetImpl of super::IHelloStarknet<ContractState> {
         // Increases the balance by the given amount.
         fn increase_balance(ref self: ContractState, amount: felt252) {
             assert_is_allowed_user();
             self.balance.write(self.balance.read() + amount);
         }
-
-        // Gets the balance. 
+        // Gets the balance.
         fn get_balance(self: @ContractState) -> felt252 {
             self.balance.read()
         }
+        // Gets the block number
+        fn get_block_number(self: @ContractState) -> u64 {
+            self.blk_nb.read()
+        }
+        // Gets the block timestamp
+        fn get_block_timestamp(self: @ContractState) -> u64 {
+            self.blk_timestamp.read()
+        }
```

Comment:
The code does not really say which block number we are getting, this is specifically the `constructor` value so it's rather 
`get_block_timestamp_at_constructor_invocation` or `get_block_timestamp_at_construction`